src/data/real/ereal.lean

```diff
@@ -695,54 +735,57 @@ bot_mul_of_neg (ereal.coe_neg'.2 h)
-  { rcases lt_trichotomy 0 y with hy|rfl|hy,
-    { simp only [bot_mul_coe_of_pos hy, to_real_bot, zero_mul] },
-    { simp only [coe_zero, mul_zero, to_real_zero] },
-    { simp only [bot_mul_coe_of_neg hy, to_real_top, to_real_bot, zero_mul] } },
-  { simp only [bot_mul_top, to_real_bot, to_real_top, mul_zero] },
-  { rcases lt_trichotomy 0 x with hx|rfl|hx,
-    { simp only [coe_mul_bot_of_pos hx, to_real_bot, mul_zero] },
-    { simp only [coe_zero, zero_mul, to_real_zero] },
-    { simp only [coe_mul_bot_of_neg hx, to_real_top, to_real_bot, mul_zero] } },
-  { simp only [← coe_mul, to_real_coe] },
-  { rcases lt_trichotomy 0 x with hx|rfl|hx,
-    { simp only [coe_mul_top_of_pos hx, to_real_top, mul_zero] },
-    { simp only [coe_zero, zero_mul, to_real_zero] },
-    { simp only [coe_mul_top_of_neg hx, to_real_top, to_real_bot, mul_zero] } },
-  { simp only [top_mul_bot, to_real_bot, mul_zero] },
-    { rcases lt_trichotomy 0 y with hy|rfl|hy,
-    { simp only [top_mul_coe_of_pos hy, to_real_top, zero_mul] },
-    { simp only [coe_zero, mul_zero, to_real_zero] },
-    { simp only [top_mul_coe_of_neg hy, to_real_top, to_real_bot, zero_mul] } },
-  { simp only [top_mul_top, to_real_top, mul_zero] }
+  -- TODO: replace with `induction using` in Lean 4, which supports multiple premises
+  with_cases
+  { apply @induction₂ (λ x y, to_real (x * y) = to_real x * to_real y) };
+    propagate_tags { try { dsimp only} },
+  case [top_zero, bot_zero, zero_top, zero_bot] { all_goals { simp only [zero_mul, mul_zero,
+                                                                         to_real_zero] } },
+  case coe_coe : x y { norm_cast },
+  case top_top { rw [top_mul_top, to_real_top, mul_zero] },
+  case top_bot { rw [top_mul_bot, to_real_top, to_real_bot, zero_mul] },
+  case bot_top { rw [bot_mul_top, to_real_bot, zero_mul] },
+  case bot_bot { rw [bot_mul_bot, to_real_top, to_real_bot, zero_mul] },
+  case pos_bot : x hx { rw [to_real_bot, to_real_coe, coe_mul_bot_of_pos hx, to_real_bot,
+                            mul_zero] },
+  case neg_bot : x hx { rw [to_real_bot, to_real_coe, coe_mul_bot_of_neg hx, to_real_top,
+                            mul_zero] },
+  case pos_top : x hx { rw [to_real_top, to_real_coe, coe_mul_top_of_pos hx, to_real_top,
+                            mul_zero] },
+  case neg_top : x hx { rw [to_real_top, to_real_coe, coe_mul_top_of_neg hx, to_real_bot,
+                            mul_zero] },
+  case top_pos : y hy { rw [to_real_top, to_real_coe, top_mul_coe_of_pos hy, to_real_top,
+                            zero_mul] },
+  case top_neg : y hy { rw [to_real_top, to_real_coe, top_mul_coe_of_neg hy, to_real_bot,
+                            zero_mul] },
+  case bot_pos : y hy { rw [to_real_bot, to_real_coe, bot_mul_coe_of_pos hy, to_real_bot,
+                            zero_mul] },
+  case bot_neg : y hy { rw [to_real_bot, to_real_coe, bot_mul_coe_of_neg hy, to_real_top,
+                            zero_mul] },
```

Comment:
```suggestion
  case pos_bot : x hx
  { rw [to_real_bot, to_real_coe, coe_mul_bot_of_pos hx, to_real_bot, mul_zero] },
  case neg_bot : x hx
  { rw [to_real_bot, to_real_coe, coe_mul_bot_of_neg hx, to_real_top, mul_zero] },
  case pos_top : x hx
  { rw [to_real_top, to_real_coe, coe_mul_top_of_pos hx, to_real_top, mul_zero] },
  case neg_top : x hx
  { rw [to_real_top, to_real_coe, coe_mul_top_of_neg hx, to_real_bot, mul_zero] },
  case top_pos : y hy
  { rw [to_real_top, to_real_coe, top_mul_coe_of_pos hy, to_real_top, zero_mul] },
  case top_neg : y hy
  { rw [to_real_top, to_real_coe, top_mul_coe_of_neg hy, to_real_bot, zero_mul] },
  case bot_pos : y hy
  { rw [to_real_bot, to_real_coe, bot_mul_coe_of_pos hy, to_real_bot, zero_mul] },
  case bot_neg : y hy
  { rw [to_real_bot, to_real_coe, bot_mul_coe_of_neg hy, to_real_top, zero_mul] },
```

Reply:
Will let the maintainer pulling the trigger decide. mathport will reformat this for mathlib4 anyway, so probably doesn't really matter.